nodejs/eks/cluster.ts

```diff
@@ -579,9 +596,68 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         nodeSecurityGroupTags: args.nodeSecurityGroupTags,
         storageClasses: userStorageClasses,
         fargateProfile: fargateProfile,
+        oidcProvider: oidcProvider,
     };
 }
 
+// Adapted from https://git.io/JvGHB
+
+// Find the root CA cert in a chain of certs.
+function findCACertificate(certificate: any, parent?: any): any {
+    if (certificate?.issuerCertificate.fingerprint !== certificate.fingerprint) {
+      return findCACertificate(certificate.issuerCertificate, certificate);
```

Comment:
I assume these are never more than 10s of levels deep so that there isn't stack overflow risk in this recursion?

Reply:
This func has been refactored to not use recursion